src/asset.rs

```diff
@@ -35,6 +35,15 @@ impl From<Asset> for AssetUnchecked {
     }
 }
 
+impl From<AssetInfo> for Asset {
+    fn from(value: AssetInfo) -> Self {
+        Self {
+            info: value,
+            amount: Uint128::zero(),
+        }
+    }
+}
```

Comment:
I am a little against implementing From for types that don't actually map onto each other. migth be better with a `to_asset(amount: Uint128)` function. Any specific reason why this is much more convenient?

Reply:
Fair enough. That's actually more useful too.